From da770222cb347e63486f0c89bf250f86a962f09f Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 26 Feb 2019 17:03:26 +0100
Subject: [PATCH] switch-api: Fix compile warnings

This moves the code to fix the following compile warnings:

drivers/net/ethernet/lantiq/switch-api/gsw_init.c: In function 'GSW_SMDIO_DataRead':
drivers/net/ethernet/lantiq/switch-api/gsw_init.c:461:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  struct core_ops *gsw_ops;
    ^
drivers/net/ethernet/lantiq/switch-api/gsw_init.c: In function 'GSW_SMDIO_DataWrite':
drivers/net/ethernet/lantiq/switch-api/gsw_init.c:486:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  struct core_ops *gsw_ops;
    ^

This only happens when CONFIG_X86_INTEL_CE2700 is not defined.
---
 drivers/net/ethernet/lantiq/switch-api/gsw_init.c | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_init.c b/drivers/net/ethernet/lantiq/switch-api/gsw_init.c
index 40050e3d27e7..2afb34e690ac 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_init.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_init.c
@@ -446,12 +446,6 @@ void gsw_w32_raw(void *cdev, short offset, u32 value)
 int GSW_SMDIO_DataRead(void *cdev, GSW_MDIO_data_t *pPar)
 {
 	int ret = 0;
-	ethsw_api_dev_t *pethdev = GSW_PDATA_GET(cdev);
-
-	if (pethdev == NULL) {
-		pr_err("%s:%s:%d", __FILE__, __func__, __LINE__);
-		return GSW_statusErr;
-	}
 #ifdef CONFIG_X86_INTEL_CE2700
 	u32 data;
 	ret = DWC_ETH_QOS_mdio_read_direct(MDIO_BUS_NUMBER_0, C45_ENABLED,
@@ -459,6 +453,12 @@ int GSW_SMDIO_DataRead(void *cdev, GSW_MDIO_data_t *pPar)
 	pPar->nData = data & 0xFFFF;
 #else
 	struct core_ops *gsw_ops;
+	ethsw_api_dev_t *pethdev = GSW_PDATA_GET(cdev);
+
+	if (pethdev == NULL) {
+		pr_err("%s:%s:%d", __FILE__, __func__, __LINE__);
+		return GSW_statusErr;
+	}
 	gsw_ops = gsw_get_swcore_ops(pethdev->parent_devid);
 	if(!gsw_ops) {
 		pr_err("%s:%s:%d", __FILE__, __func__, __LINE__);
@@ -473,17 +473,18 @@ int GSW_SMDIO_DataRead(void *cdev, GSW_MDIO_data_t *pPar)
 int GSW_SMDIO_DataWrite(void *cdev, GSW_MDIO_data_t *pPar)
 {
 	int ret = 0;
+#ifdef CONFIG_X86_INTEL_CE2700
+	ret = DWC_ETH_QOS_mdio_write_direct(MDIO_BUS_NUMBER_0, C45_ENABLED,
+					    MDIO_ADDR_LANTIQ, MMD_DISABLED, pPar->nAddressReg & 0x1F, pPar->nData & 0xFFFF);
+#else
+	struct core_ops *gsw_ops;
 	ethsw_api_dev_t *pethdev = GSW_PDATA_GET(cdev);
 
 	if (pethdev == NULL) {
 		pr_err("%s:%s:%d", __FILE__, __func__, __LINE__);
 		return GSW_statusErr;
 	}
-#ifdef CONFIG_X86_INTEL_CE2700
-	ret = DWC_ETH_QOS_mdio_write_direct(MDIO_BUS_NUMBER_0, C45_ENABLED,
-					    MDIO_ADDR_LANTIQ, MMD_DISABLED, pPar->nAddressReg & 0x1F, pPar->nData & 0xFFFF);
-#else
-	struct core_ops *gsw_ops;
+
 	gsw_ops = gsw_get_swcore_ops(pethdev->parent_devid);
 	if(!gsw_ops) {
 		pr_err("%s:%s:%d", __FILE__, __func__, __LINE__);
